### DISCUSSION TITLE
Auto-complete / preview for registered functions?

### DISCUSSION THREAD

**Discussion by Rezan7CC** [doonv/bevy_dev_console#24] [Ideas]
Hey!

I just started trying to use this console for the simple purpose of supporting dev-cheats similar to this and I enjoyed using it so far:
```
fn set_god_mode_enabled(enabled: u8) {
    println!("God mode enabled: {}", enabled);
}

pub fn console_environment() -> Environment {
    let mut environment = Environment::default();
    
    environment.register_fn("godmode", set_god_mode_enabled);

    environment
}
```

However something that could be incredibly useful if there is a larger amount of registered functions/commands would be some form of preview and perhaps auto-complete for commands. Did you ever consider adding that?

**Comment by doonv** [comment]
I did consider adding that but I ended up not doing it because there were other things I wanted to work on first.

**Reply by Rezan7CC** [reply]
Fair enough! Well I will be using the console in the long-term either way.
Could be a cool addition in case you ever plan to add more to it.

**Comment by Rezan7CC** [comment]
![image](https://github.com/user-attachments/assets/43d21cee-58c4-4fea-93c5-323ce7ac6809)

Oh wow!
